src/android/io/branch/BranchSDK.java

```diff
@@ -404,7 +421,9 @@ private void showShareSheet(JSONObject options, JSONObject controlParams) throws
 
         BranchLinkProperties linkProperties = createLinkProperties(options, controlParams);
 
-        this.branchObj.showShareSheet(this.activity, linkProperties, shareSheetStyle, new ShowShareSheetListener());
+        BranchUniversalObject branchObj = (BranchUniversalObject)this.branchObjects.get(instanceIdx);
+
+        branchObj.showShareSheet(this.activity, linkProperties, shareSheetStyle, new ShowShareSheetListener());
```

Comment:
As per my comment on line 50, I'd expect the listener to take in the instanceId or pull in a callback to call from the instanceId, instead of just (later) calling "onShareLinkDialogLaunched.sendPluginResult(result);"